### PR TITLE
Implement simulation step and write scenario function

### PR DESCRIPTION
**Pull request by EdenTewelde**
Please have a look at my approach to issue #23 and let me know about your thoughts. I am not sure if I completely understood this task.

### CODE REVIEW

File: src/main/kotlin/Network.kt [hunk 1]
```diff
@@ -14,11 +14,14 @@ class Network(val capacity : Int) {
         }
     }
 
-    fun analyzeNetwork() {
+    fun analyzeNetwork() : String {
         val overCapacity = this.overCapacity()
         for (car in this.listOfCarsDriving) {
             car.delayed = overCapacity
+            val isDelayed = car.delayed
+            return "$isDelayed"
```
Comment (nipafx) [inline src/main/kotlin/Network.kt:22]: When you return inside a loop, the loop stops immediately. I don't think this is what you intended. :smiley:  I actually see no reason to change this function.

File: src/main/kotlin/scenario.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+fun simulationStep(numberOfCars : Int, capacity : Int) {
+    val road = Network(capacity)
+    val cars = 1..numberOfCars
+    for (car in cars) {
+        val car = Car(true, road)
+        println("Does this car want to drive? " + car.wantsToDrive)
+        road.overCapacity()
+        println("Is this car ($car) delayed? " + road.analyzeNetwork())
```
Comment (nipafx) [inline src/main/kotlin/scenario.kt:12]: `analyzeNetwork` is the function that runs the simulation - here you call it once per car. I don't think that's quite what you wanted.

**Review by nipafx — COMMENTED** [review]
The code gives the impression that you're a little unclear on what exactly the goal is. Maybe it makes sense to go back to #23 and discuss what you want to do.

File: src/main/kotlin/Network.kt (hunk 1 above)
Comment (EdenTewelde) [reply src/main/kotlin/Network.kt:22]: Thanks, I didn't know that

File: src/main/kotlin/scenario.kt (hunk 2 above)
Comment (EdenTewelde) [reply src/main/kotlin/scenario.kt:12]: Right, but I need to know for every car whether it is delayed or is able to drive

Comment (nipafx) [reply src/main/kotlin/scenario.kt:12]: Given a list of cars, `analyzeNetwork` sets for each whether it is delayed or not. Once That is done, you can simply loop over that list and print for each car, which value got set.

**Pull request closed without merging**